migration_original/ODS1Stage/tables/Base/ProviderToProviderSubType/spu_translated_ProviderToProviderSubType.sql

```diff
@@ -35,21 +36,31 @@ begin
 ---------------------------------------------------------     
 
 --- select Statement
-select_statement := $$ select distinct
+select_statement :=     $$ 
+                        with cte_providersubtype as (
+                            select
+                                p.ref_provider_code as providercode,
+                                to_varchar(json.value:PROVIDER_SUB_TYPE_CODE) as providersubtype_ProviderSubTypeCode,
+                                to_varchar(json.value:PROVIDER_SUB_TYPE_RANK) as providersubtype_ProviderSubTypeRank,
+                                to_varchar(json.value:DATA_SOURCE_CODE) as providersubtype_SourceCode,
+                                to_timestamp_ntz(json.value:UPDATED_DATETIME) as providersubtype_LastUpdateDate
+                            from $$||mdm_db||$$.mst.provider_profile_processing as p,
+                            lateral flatten(input => p.PROVIDER_PROFILE:PROVIDER_SUB_TYPE) as json
+                        )
+                        
+                        select distinct
                             p.providerid,
                             pst.providersubtypeid,
                             ifnull(json.providersubtype_SourceCode, 'Profisee') as SourceCode,
                             json.providersubtype_ProviderSubTypeRank as ProviderSubTypeRank,
                             2147483647 as ProviderSubTypeRankCalculated,
                             ifnull(json.providersubtype_LastUpdateDate, current_timestamp()) as LastUpdateDate
-                        from raw.vw_PROVIDER_PROFILE as JSON
-                            left join base.provider as P on p.providercode = json.providercode
-                            join base.providersubtype as PST on pst.providersubtypecode = json.providersubtype_ProviderSubTypeCode
-                        where
-                            PROVIDER_PROFILE is not null
-                            and ProviderSubType_ProviderSubTypeCode is not null
-                            and ProviderID is not null 
-                        qualify row_number() over( partition by ProviderID, ifnull(ProviderSubType_ProviderSubTypeCode, 'ALT') order by CREATE_DATE desc) = 1 $$;
+                        from cte_providersubtype as json
+                        join base.provider as p on p.providercode = json.providercode
+                        join base.providersubtype as pst on pst.providersubtypecode = json.providersubtype_ProviderSubTypeCode
+                        where json.providersubtype_ProviderSubTypeCode is not null
```

Comment:
you can add this where clause in the select above ( in the cte )

Reply:
I don't see the benefit of doing this. I ran the select statement a couple of times and there weren't any performance gains, and in terms of organizational purposes it makes more sense to have the CTE have the raw flattened data and the filters below in the final select statement.